src/lib/utils/async.js

```diff
@@ -31,15 +31,45 @@ export const promisifyGun = async callback =>
     const onAck = once(ack => {
       const { err } = ack
 
+      // no err - resolve
       if (!err) {
         resolve(ack)
+        return
       }
 
-      const exception = new Error(err)
+      // if ack.err is an JS error - rejecting with it
+      let exception = err
 
+      // otherwise creating a new Error object
+      if (!isError(err)) {
+        // by default we'll show some generic message
+        let message = 'Unexpected GUN error during write / encrypt operation'
+        const { code, errno } = err || {}
+
+        // if ack.err is a string we'll use it as the error message
+        if (isString(err)) {
+          message = err
+        } else if (isObject(err) && code && errno) {
+          // also err could be an object with { code, errno, syscall, path } shape
+          // in this case we'll show the message with errno & code
+          message = `GUN error ${errno}: ${code}`
```

Comment:
here we lose the data in error object that gun returns. maybe just json stringify it.
`Gun error ${JSON.stringify(err)`

Reply:
1. there just 2 more fields (path & syscall). Path is too long but ok, i'll add it
2. we don't loose anything as the ack object is stored as the exception.ack so it could be used when exception is caught